src/modules/cpu/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,1814 @@
+#if __AVX2__
+                    // process alignedLength number of columns in each row
+                    for (; vectorLoopCount < alignedLength; vectorLoopCount += 8)
+                    {
+                        __m256 pResultPln[3];
+                        for (int c = 0; c < 3; c++)
+                        {
+                            __m256 pRow[14];
+                            rpp_load_filter_7x7_pln_host(pRow, srcPtrTemp[c], rowKernelLoopLimit);
+                            pResultPln[c] = avx_p0;
+                            for (int k = 0; k < 7; k++)
+                            {
+                                __m256 pTemp[7];
+                                Rpp32s filterIndex =  k * 7;
+                                Rpp32s rowIndex = k * 2;
+
+                                pTemp[0] = _mm256_mul_ps(pRow[rowIndex], pFilter[filterIndex]);
+                                pTemp[1] = _mm256_mul_ps(_mm256_permutevar8x32_ps(_mm256_blend_ps(pRow[rowIndex], pRow[rowIndex + 1], 1), avx_pxMaskRotate0To1), pFilter[filterIndex + 1]);
+                                pTemp[2] = _mm256_mul_ps(_mm256_permutevar8x32_ps(_mm256_blend_ps(pRow[rowIndex], pRow[rowIndex + 1], 3), avx_pxMaskRotate0To2), pFilter[filterIndex + 2]);
+                                pTemp[3] = _mm256_mul_ps(_mm256_permutevar8x32_ps(_mm256_blend_ps(pRow[rowIndex], pRow[rowIndex + 1], 7), avx_pxMaskRotate0To3), pFilter[filterIndex + 3]);
+                                pTemp[4] = _mm256_mul_ps(_mm256_permutevar8x32_ps(_mm256_blend_ps(pRow[rowIndex], pRow[rowIndex + 1], 15), avx_pxMaskRotate0To4), pFilter[filterIndex + 4]);
+                                pTemp[5] = _mm256_mul_ps(_mm256_permutevar8x32_ps(_mm256_blend_ps(pRow[rowIndex], pRow[rowIndex + 1], 31), avx_pxMaskRotate0To5), pFilter[filterIndex + 5]);
+                                pTemp[6] = _mm256_mul_ps(_mm256_permutevar8x32_ps(_mm256_blend_ps(pRow[rowIndex], pRow[rowIndex + 1], 63), avx_pxMaskRotate0To6), pFilter[filterIndex + 6]);
+                                pResultPln[c] = _mm256_add_ps(pResultPln[c], _mm256_add_ps(_mm256_add_ps(pTemp[0], _mm256_add_ps(pTemp[1], pTemp[2])), _mm256_add_ps(_mm256_add_ps(pTemp[3], pTemp[4]), _mm256_add_ps(pTemp[5], pTemp[6]))));
+                            }
+                            increment_row_ptrs(srcPtrTemp[c], kernelSize, 8);
+                        }
+                        // convert result from pln to pkd format and store in output buffer
+                        if constexpr (std::is_same<T, Rpp32f>::value)
+                            rpp_store24_f32pln3_to_f32pkd3_avx(dstPtrTemp, pResultPln);
+                        else if constexpr (std::is_same<T, Rpp16f>::value)
+                            rpp_store24_f32pln3_to_f16pkd3_avx(dstPtrTemp, pResultPln);
+                        else if constexpr (std::is_same<T, Rpp8u>::value)
+                            rpp_simd_store(rpp_store24_f32pln3_to_u8pkd3_avx, dstPtrTemp, pResultPln);
```

Comment:
For some we are using rpp_simd_store and for some it is not used
Please either add/remove this for all instances

Reply:
Done